recipes-bsp/u-boot/files/0009-board-siemens-iot2050-Generalize-the-fdt-fixup.patch

```diff
@@ -0,0 +1,111 @@
+have immediate effect for other variant, however this makes the future
+variant more easier to apply fdt fixups.
+
+Signed-off-by: Baocheng Su <baocheng.su@siemens.com>
+---
+ board/siemens/iot2050/board.c | 32 +++++++++++++++++++-------------
+ doc/board/siemens/iot2050.rst |  7 +++++++
+ 2 files changed, 26 insertions(+), 13 deletions(-)
+
+diff --git a/board/siemens/iot2050/board.c b/board/siemens/iot2050/board.c
+index d6228b9bd26b..32f5280125c0 100644
+--- a/board/siemens/iot2050/board.c
++++ b/board/siemens/iot2050/board.c
+@@ -245,23 +245,14 @@ void set_board_info_env(void)
+ 	env_save();
+ }
+ 
+-static void m2_overlay_prepare(void)
++static void do_overlay_prepare(const char *overlay_path)
+ {
+ #if defined(CONFIG_OF_LIBFDT) && defined(CONFIG_OF_BOARD_SETUP)
+-	const char *overlay_path;
+ 	void *overlay;
+ 	u64 loadaddr;
+ 	ofnode node;
+ 	int ret;
+ 
+-	if (connector_mode == BKEY_PCIEX2)
+-		return;
+-
+-	if (connector_mode == BKEY_PCIE_EKEY_PCIE)
+-		overlay_path = "/fit-images/bkey-ekey-pcie-overlay";
+-	else
+-		overlay_path = "/fit-images/bkey-usb3-overlay";
+-
+ 	node = ofnode_path(overlay_path);
+ 	if (!ofnode_valid(node))
+ 		goto fit_error;
+@@ -288,6 +279,21 @@ fit_error:
+ #endif
+ }
+ 
++static void m2_overlay_prepare(void)
++{
++	const char *overlay_path;
++
++	if (connector_mode == BKEY_PCIEX2)
++		return;
++
++	if (connector_mode == BKEY_PCIE_EKEY_PCIE)
++		overlay_path = "/fit-images/bkey-ekey-pcie-overlay";
++	else
++		overlay_path = "/fit-images/bkey-usb3-overlay";
++
++	do_overlay_prepare(overlay_path);
++}
++
+ static void m2_connector_setup(void)
+ {
+ 	ulong m2_manual_config = env_get_ulong("m2_manual_config", 10,
+@@ -466,7 +472,7 @@ int board_late_init(void)
+ }
+ 
+ #if defined(CONFIG_OF_LIBFDT) && defined(CONFIG_OF_BOARD_SETUP)
+-static void m2_fdt_fixup(void *blob)
++static void variants_fdt_fixup(void *blob)
+ {
+ 	void *overlay_copy = NULL;
+ 	void *fdt_copy = NULL;
+@@ -506,14 +512,14 @@ cleanup:
+ 	return;
+ 
+ fixup_error:
+-	pr_err("Could not apply M.2 device tree overlay\n");
++	pr_err("Could not apply device tree overlay\n");
+ 	goto cleanup;
+ }
+ 
+ int ft_board_setup(void *blob, struct bd_info *bd)
+ {
+ 	if (board_is_m2())
+-		m2_fdt_fixup(blob);
++		variants_fdt_fixup(blob);
+ 
+ 	return 0;
+ }
+diff --git a/doc/board/siemens/iot2050.rst b/doc/board/siemens/iot2050.rst
+index c6bb06750584..21fb9d694ffe 100644
+--- a/doc/board/siemens/iot2050.rst
++++ b/doc/board/siemens/iot2050.rst
+@@ -167,3 +167,10 @@ the U-Boot environment variable "m2_manual_config" to select the mode manually:
+         E-key: PCIe, USB 2.0
+ "2"  -  B-key: USB 3.0,
+         E-key: PCIe, USB 2.0
++
++Device tree overlay fixup
++-------------------------
++
++In rare cases, you may need to add your own device tree overlay, please take use
++of the **do_overlay_prepare** and **variants_fdt_fixup** from
++**board/siemens/iot2050/board.c** to add overlay fixup.
```

Comment:
Is that really a use case we want to document?

Reply:
And this one.

Reply:
Fixed